Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs

```diff
@@ -45,13 +49,19 @@ public sealed class MicrowaveSystem : EntitySystem
         [Dependency] private readonly RecipeManager _recipeManager = default!;
         [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
         [Dependency] private readonly SharedAudioSystem _audio = default!;
+        [Dependency] private readonly LightningSystem _lightning = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+        [Dependency] private readonly ExplosionSystem _explosion = default!;
+        [Dependency] private readonly SharedDestructibleSystem _destruction = default!;
         [Dependency] private readonly SharedContainerSystem _sharedContainer = default!;
         [Dependency] private readonly SolutionContainerSystem _solutionContainer = default!;
         [Dependency] private readonly TagSystem _tag = default!;
         [Dependency] private readonly TemperatureSystem _temperature = default!;
         [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
         [Dependency] private readonly HandsSystem _handsSystem = default!;
         [Dependency] private readonly SharedItemSystem _item = default!;
+        private TimeSpan _targetTime = TimeSpan.Zero;
```

Comment:
You should use a floating frameTime instead of a TimeSpan, see `GasAnalyzerSystem.cs` for an example. There is a strong preference for the accumulated frametime to be stored in the component instead of the system.

Reply:
The microwave's `CookTimeRemaining` is also a good example of how to do this.